cmd/asset-syncer/utils.go

```diff
@@ -145,6 +148,68 @@ func (r *HelmRepo) Repo() *models.RepoInternal {
 	return r.RepoInternal
 }
 
+func satisfy(chartInput map[string]interface{}, rule *apprepov1alpha1.FilterRuleSpec) (bool, error) {
+	query, err := gojq.Parse(rule.JQ)
+	if err != nil {
+		return false, fmt.Errorf("Unable to parse JQuery: %v", err)
+
+	}
+	varNames := []string{}
+	varValues := []interface{}{}
+	for name, val := range rule.Variables {
+		varNames = append(varNames, name)
+		varValues = append(varValues, val)
+	}
+	code, err := gojq.Compile(
+		query,
+		gojq.WithVariables(varNames),
+	)
+	if err != nil {
+		return false, fmt.Errorf("Unable to compile JQuery: %v", err)
+	}
+
+	res, _ := code.Run(chartInput, varValues...).Next()
```

Comment:
So everything above this line in this function has been compiling the jq query, independent of any chartInput. Yet this function is called during an iteration of charts, so we're doing all of the above to compile/prepare the jq query each time, for each chart. It seems odd that we're not compiling the `code` in a separate function and calling that just once below *before* iterating the charts?

Reply:
Great, thanks for the update to compile it once before the iteration!